File: AMS.API/DTO/BatchMaster/BatchMaster.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace AMS.API.DTO.BatchMaster
{
    public class BatchMaster
    {
        public Int64 Id { get; set; }

        [Required]
        [StringLength(10)]
        public string BatchId { get; set; }

        public string BatchName { get; set; }

        [JsonIgnore]
        public DateTime? CreatedDate { get; set; }

        public Int32 CreatedBy { get; set; }

        [JsonIgnore]
        public DateTime? ModifiedDate { get; set; }

        public Int32 ModifiedBy { get; set; }

        public bool IsActive { get; set; }

        /// <summary>
        /// All other batch details except BatchId and BatchName
        /// </summary>
        public BatchDetail BatchDetails { get; set; }
    }
}

File: AMS.Web/src/app/models/batch-master/batch-master.ts
import { BatchDetails } from "./batch-details";
import { BatchAllocationDetail } from "../batch-allocation/batch-allocation-detail";

export class BatchMaster 
{
    constructor(){
        this.BatchDetails = new BatchDetails();
        this.BatchAllocationDetails = new BatchAllocationDetail();
    };

    Id: number;
    BatchId: string = null;
    BatchName?: string = null;

    //All other batch details except BatchId and BatchName
    BatchDetails : BatchDetails = null;

    //Batch-Allocation details 
    BatchAllocationDetails: BatchAllocationDetail

    CreatedDate? : Date = null;
    CreatedBy? : number = null;
    ModifiedDate? : Date = null;
    ModifiedBy? : number = null;
    IsActive : boolean = true;
}
